File: swap_flutter/lib/core/constants/api_paths.dart
/// API Paths Module
/// 
/// This module defines all API endpoints used in the application.
/// Adapted from Expo apiPaths.ts to maintain the same structure and organization.
/// 
/// - DO NOT include 'api/' or '/api/' at the beginning of paths
/// - DO NOT include 'v1/' or '/v1/' at the beginning of paths  
/// - The apiClient already adds the base URL with '/api/v1/' automatically
/// - All paths in this file should start directly with the resource name
/// 
/// CORRECT:   '/auth/login'
/// INCORRECT: '/api/auth/login' or '/api/v1/auth/login'

/// Authentication endpoints
class AuthPaths {
  static const String login = '/auth/login';
  static const String pinLogin = '/auth/pin-login';
  static const String businessLogin = '/auth/business/login';
  static const String businessRegister = '/auth/business/register';
  static const String logout = '/auth/logout';
  static const String refresh = '/auth/refresh';
  static const String profile = '/auth/profile';
  static const String me = '/auth/me';
  static const String verify = '/auth/verify-token';
  static const String register = '/auth/register';
  static const String forgotPassword = '/auth/forgot-password';
  static const String resetPassword = '/auth/reset-password';
  static const String phoneSignin = '/auth/phone-signin';
  static const String verifyPhone = '/auth/verify-phone';
  static const String registerPhone = '/auth/register-phone';
  static const String checkPhone = '/auth/check-phone';
  static const String completeProfile = '/auth/complete-profile';
  static const String storePasscode = '/auth/store-passcode';
  static const String verifyEmail = '/auth/verify-email';
  static const String resendEmailCode = '/auth/resend-email-code';
  static const String verifyResetCode = '/auth/verify-reset-code';
}

/// User management endpoints
class UserPaths {
  static String profile(String id) => '/profiles/$id';
  static const String search = '/users/search';
  static const String discover = '/users/discover';
  static const String preferences = '/users/preferences';
}

/// KYC (Know Your Customer) endpoints
class KycPaths {
  static const String status = '/kyc/verification/status';
  static const String requirements = '/kyc/verification/requirements';
  static const String documents = '/kyc/verification/documents';
  static const String submitDocuments = '/kyc/verification/documents';
  static const String upload = '/kyc/upload';
  static const String verifyPhone = '/kyc/phone/verify';
  static const String verifyEmail = '/kyc/email/verify';
  static const String biometricSetup = '/kyc/biometric/setup';
  static const String requestPhoneChange = '/kyc/phone/request-change';
  static const String selfieComplete = '/kyc/selfie/complete';
  
  /// Admin endpoints for KYC
  static const String adminPending = '/kyc/admin/pending';
  static String adminDocument(String id) => '/kyc/admin/documents/$id';
  static String adminReview(String id) => '/kyc/admin/documents/$id/review';
}

/// Reference data endpoints
class ReferenceDataPaths {
  static const String countries = '/countries';
}

/// Account management endpoints
class AccountPaths {
  static const String list = '/accounts';
  static const String balances = '/accounts/balances';
  static String details(String id) => '/accounts/$id';
  static String update(String id) => '/accounts/$id';
  static String setPrimary(String id) => '/accounts/$id';
  static String balance(String id) => '/accounts/$id/balance';
  static String statement(String id) => '/accounts/$id/statement';
  static String byEntity(String? entityId) => 
    entityId != null ? '/accounts/entity/$entityId' : '/accounts';
}

/// Transaction endpoints
class TransactionPaths {
  static const String list = '/transactions';
  static String details(String id) => '/transactions/$id';
  static const String create = '/transactions';
  static String interaction(String interactionId) => '/transactions/interaction/$interactionId';
  static const String accounts = '/transactions/accounts';
  static String accountDetails(String id) => '/transactions/accounts/$id';
  static const String currencies = '/transactions/currencies';
  static String currencyDetails(String id) => '/transactions/currencies/$id';
  static const String requests = '/transactions/requests';
  static String requestDetails(String id) => '/transactions/requests/$id';
  static String totalSent(String senderId, String recipientId) => 
    '/transactions/total-sent/$senderId/$recipientId';
  static const String direct = '/transactions/direct';
}

/// Wallet endpoints (NEW - replaces account balance endpoints)
class WalletPaths {
  /// Get wallets by account (replaces account balances)
  static String byAccount(String accountId) => '/wallets/account/$accountId';
  
  /// Get wallets by entity (user/business)
  static String byEntity(String entityId) => '/wallets/entity/$entityId';
  
  /// Individual wallet operations
  static String details(String walletId) => '/wallets/$walletId';
  static String credit(String walletId) => '/wallets/$walletId/credit';
  static String debit(String walletId) => '/wallets/$walletId/debit';
  
  /// Get or create wallet for specific account and currency
  static String getOrCreate(String accountId, String currencyId) => 
    '/wallets/account/$accountId/currency/$currencyId';
}

/// Interaction endpoints
class InteractionPaths {
  static const String list = '/interactions';
  static const String recent = '/interactions/recent';
  static String details(String id) => '/interactions/$id';
  static const String create = '/interactions';
  static String direct(String profileId) => '/interactions/direct/$profileId';
  static String messages(String interactionId) => '/interactions/$interactionId/messages';
  static String timeline(String interactionId) => '/interactions/$interactionId/timeline';
}

/// Business endpoints
class BusinessPaths {
  static const String profiles = '/business/profiles';
  static String profileDetails(String id) => '/business/profiles/$id';
  static const String locations = '/business/locations';
  static String locationDetails(String id) => '/business/locations/$id';
  static const String team = '/business/team';
  static String teamMember(String id) => '/business/team/$id';
  static const String offers = '/business/offers';
  static String offerDetails(String id) => '/business/offers/$id';
}

/// Notification endpoints
class NotificationPaths {
  static const String list = '/notifications';
  static String details(String id) => '/notifications/$id';
  static const String preferences = '/notifications/preferences';
  static const String devices = '/notifications/devices';
}

/// Entity endpoints
class EntityPaths {
  static const String list = '/entities';
  static String get(String id) => '/entities/$id';
  static String byReference(String type, String id) => '/entities/reference/$type/$id';
  static String resolve(String id) => '/entities/resolve/$id';
  static const String search = '/entities/search';
  static String update(String id) => '/entities/$id';
}

/// Message endpoints
class MessagePaths {
  static const String create = '/messages/direct';
  static String listForInteraction(String interactionId) => 
    '/interactions/$interactionId/messages';
}

/// Unified Search endpoints
class SearchPaths {
  static const String all = '/search'; // GET /api/v1/search?query=...
}

/// Combined export of all API paths
/// This provides a centralized access point to all endpoint definitions
class ApiPaths {
  static final auth = AuthPaths();
  static final user = UserPaths();
  static final kyc = KycPaths();
  static final referenceData = ReferenceDataPaths();
  static final account = AccountPaths();
  static final wallet = WalletPaths(); // NEW: Wallet endpoints
  static final transaction = TransactionPaths();
  static final interaction = InteractionPaths();
  static final message = MessagePaths();
  static final business = BusinessPaths();
  static final notification = NotificationPaths();
  static final entities = EntityPaths();
  static final search = SearchPaths();
}

/// Type-safe endpoint builder for parameterized paths
class EndpointBuilder {
  /// Helper method to build parameterized URLs safely
  static String buildPath(String template, Map<String, String> params) {
    String result = template;
    params.forEach((key, value) {
      result = result.replaceAll(':$key', value);
    });
    return result;
  }
  
  /// Validate that all required parameters are provided
  static void validateParams(String template, Map<String, String> params) {
    final RegExp paramRegex = RegExp(r':(\w+)');
    final matches = paramRegex.allMatches(template);
    
    for (final match in matches) {
      final paramName = match.group(1)!;
      if (!params.containsKey(paramName)) {
        throw ArgumentError('Missing required parameter: $paramName');
      }
    }
  }
}

File: swap-expo/_app/_api/apiPaths.ts
/**
 * API Paths Module
 * 
 * This module defines all API endpoints used in the application.
 * - DO NOT include 'api/' or '/api/' at the beginning of paths
 * - DO NOT include 'v1/' or '/v1/' at the beginning of paths
 * - The apiClient already adds the base URL with '/api/v1/' automatically
 * - All paths in this file should start directly with the resource name
 * 
 * CORRECT:   '/auth/login'
 * INCORRECT: '/api/auth/login' or '/api/v1/auth/login'
 */

/**
 * Authentication endpoints
 */
export const AUTH_PATHS = {
  LOGIN: '/auth/login',
  PIN_LOGIN: '/auth/pin-login',
  BUSINESS_LOGIN: '/auth/business/login',
  BUSINESS_REGISTER: '/auth/business/register',
  LOGOUT: '/auth/logout',
  REFRESH: '/auth/refresh',
  PROFILE: '/auth/profile',
  ME: '/auth/me',
  VERIFY: '/auth/verify-token',
  REGISTER: '/auth/register',
  FORGOT_PASSWORD: '/auth/forgot-password',
  RESET_PASSWORD: '/auth/reset-password',
  PHONE_SIGNIN: '/auth/phone-signin',
  VERIFY_PHONE: '/auth/verify-phone',
  REGISTER_PHONE: '/auth/register-phone',
  CHECK_PHONE: '/auth/check-phone',
  CHECK_USERNAME: '/auth/check-username',
  COMPLETE_PROFILE: '/auth/complete-profile',
  COMPLETE_BUSINESS_PROFILE: '/auth/complete-business-profile',
  STORE_PASSCODE: '/auth/store-passcode',
  // Email verification disabled for Haiti market - keeping for future use
  // VERIFY_EMAIL: '/auth/verify-email',
  // RESEND_EMAIL_CODE: '/auth/resend-email-code',
  VERIFY_RESET_CODE: '/auth/verify-reset-code',
  // Multi-account management (Instagram-style)
  AVAILABLE_PROFILES: '/auth/available-profiles',
  SWITCH_PROFILE: '/auth/switch-profile',
  // Generic PIN check for any profile type (personal or business)
  CHECK_PROFILE_PIN: (profileId: string) => `/auth/profiles/${profileId}/check-pin`,
  // Session management (Active Devices)
  SESSIONS: '/auth/sessions',
  SESSION_REVOKE: (sessionId: string) => `/auth/sessions/${sessionId}`,
  SESSIONS_REVOKE_ALL: '/auth/sessions',
  // Biometric authentication (secure device-based login)
  BIOMETRIC_ENABLE: '/auth/biometric/enable',
  BIOMETRIC_LOGIN: '/auth/biometric/login',
  BIOMETRIC_DISABLE: '/auth/biometric',
  BIOMETRIC_STATUS: '/auth/biometric/status',
  BIOMETRIC_DEVICES: '/auth/biometric/devices',
};

/**
 * User management endpoints
 */
export const USER_PATHS = {
  PROFILE: (id: string) => `/profiles/${id}`,
  SEARCH: '/users/search',
  DISCOVER: '/users/discover',
  PREFERENCES: '/users/preferences',
};

/**
 * KYC (Know Your Customer) endpoints
 * Professional RESTful design following Fortune 500 standards
 */
export const KYC_PATHS = {
  STATUS: '/kyc/verification-status',
  REQUIREMENTS: '/kyc/verification-requirements',

  // Clean RESTful Document Management Endpoints
  DOCUMENTS: '/kyc/documents',                              // POST - Upload documents, GET - List documents
  DOCUMENT_DETAILS: (documentId: string) => `/kyc/documents/${documentId}`,  // GET - Get specific document
  DOCUMENT_UPDATE: (documentId: string) => `/kyc/documents/${documentId}`,   // PUT - Update/replace document

  // Clean RESTful Selfie Endpoints
  SELFIE: '/kyc/selfie',                                    // POST - Upload selfie, GET - Get selfie

  // Other KYC endpoints with hyphen convention
  VERIFY_PHONE: '/kyc/phone-verify',
  // VERIFY_EMAIL: '/kyc/email-verify', // Email verification disabled for Haiti market
  // BIOMETRIC_SETUP: removed - biometric is local-only, not tracked in backend
  REQUEST_PHONE_CHANGE: '/kyc/phone-request-change',
  ADMIN: {
    PENDING: '/kyc/admin/pending',
    DOCUMENT: (id: string) => `/kyc/admin/documents/${id}`,
    REVIEW: (id: string) => `/kyc/admin/documents/${id}/review`,
  },
};

/**
 * Reference data endpoints
 */
export const REFERENCE_DATA_PATHS = {
  COUNTRIES: '/countries',
  CURRENCIES: '/currencies',
};

/**
 * Account management endpoints
 */
export const ACCOUNT_PATHS = {
  LIST: '/accounts',
  BALANCES: '/accounts/balances',
  DETAILS: (id: string) => `/accounts/${id}`,
  UPDATE: (id: string) => `/accounts/${id}`,
  SET_PRIMARY: (id: string) => `/accounts/${id}`,
  BALANCE: (id: string) => `/accounts/${id}/balance`,
  STATEMENT: (id: string) => `/accounts/${id}/statement`,
  BY_ENTITY: (entityId: string | null | undefined) => 
    entityId ? `/accounts/entity/${entityId}` : '/accounts',
};

/**
 * Transaction endpoints
 */
export const TRANSACTION_PATHS = {
  LIST: '/transactions',
  DETAILS: (id: string) => `/transactions/${id}`,
  CREATE: '/transactions',
  INTERACTION: (interactionId: string) => `/transactions/interaction/${interactionId}`,
  ACCOUNTS: '/transactions/accounts',
  ACCOUNT_DETAILS: (id: string) => `/transactions/accounts/${id}`,
  CURRENCIES: '/transactions/currencies',
  CURRENCY_DETAILS: (id: string) => `/transactions/currencies/${id}`,
  REQUESTS: '/transactions/requests',
  REQUEST_DETAILS: (id: string) => `/transactions/requests/${id}`,
  TOTAL_SENT: (fromEntityId: string, toEntityId: string) => `/transactions/total-sent/${fromEntityId}/${toEntityId}`,
  DIRECT: '/transactions/direct',
};

/**
 * Wallet endpoints (NEW - replaces account balance endpoints)
 */
export const WALLET_PATHS = {
  // Get wallets by account (replaces account balances)
  BY_ACCOUNT: (accountId: string) => `/wallets/account/${accountId}`,
  
  // Get wallets by entity (user/business)
  BY_ENTITY: (entityId: string) => `/wallets/entity/${entityId}`,
  
  // Individual wallet operations
  DETAILS: (walletId: string) => `/wallets/${walletId}`,
  CREDIT: (walletId: string) => `/wallets/${walletId}/credit`,
  DEBIT: (walletId: string) => `/wallets/${walletId}/debit`,
  
  // Get or create wallet for specific account and currency
  GET_OR_CREATE: (accountId: string, currencyId: string) => `/wallets/account/${accountId}/currency/${currencyId}`,
};

/**
 * Interaction endpoints
 */
export const INTERACTION_PATHS = {
  LIST: '/interactions',
  RECENT: '/interactions/recent',
  ARCHIVED: '/interactions/archived',
  DETAILS: (id: string) => `/interactions/${id}`,
  DELETE: (id: string) => `/interactions/${id}`,
  UNARCHIVE: (id: string) => `/interactions/${id}/unarchive`,
  CREATE: '/interactions',
  DIRECT: (profileId: string) => `/interactions/direct/${profileId}`,
  MESSAGES: (interactionId: string) => `/interactions/${interactionId}/messages`,
  TIMELINE: (interactionId: string) => `/interactions/${interactionId}/timeline`,
};

/**
 * Business endpoints
 */
export const BUSINESS_PATHS = {
  PROFILES: '/business/profiles',
  PROFILE_DETAILS: (id: string) => `/business/profiles/${id}`,
  LOCATIONS: '/business/locations',
  LOCATION_DETAILS: (id: string) => `/business/locations/${id}`,
  TEAM: '/business/team',
  TEAM_MEMBER: (id: string) => `/business/team/${id}`,
  OFFERS: '/business/offers',
  OFFER_DETAILS: (id: string) => `/business/offers/${id}`,
  // Business access PIN endpoints
  CHECK_PIN: (businessId: string) => `/businesses/${businessId}/check-pin`,
  SET_PIN: (businessId: string) => `/businesses/${businessId}/set-pin`,
  REMOVE_PIN: (businessId: string) => `/businesses/${businessId}/pin`,
};

/**
 * Notification endpoints
 */
export const NOTIFICATION_PATHS = {
  LIST: '/notifications',
  DETAILS: (id: string) => `/notifications/${id}`,
  PREFERENCES: '/notifications/preferences',
  DEVICES: '/notifications/devices',
};

/**
 * Entity endpoints
 */
export const ENTITY_PATHS = {
  LIST: '/entities',
  GET: (id: string) => `/entities/${id}`,
  BY_REFERENCE: (type: 'profile' | 'business' | 'system', id: string) => `/entities/reference/${type}/${id}`,
  RESOLVE: (id: string) => `/entities/resolve/${id}`,
  SEARCH: '/entities/search',
  UPDATE: (id: string) => `/entities/${id}`,
};

/**
 * Message endpoints
 */
export const MESSAGE_PATHS = {
  CREATE: '/messages/direct',
  LIST_FOR_INTERACTION: (interactionId: string) => `/interactions/${interactionId}/messages`,
};

/**
 * Unified Search endpoints
 */
export const SEARCH_PATHS = {
  ALL: '/search', // GET /api/v1/search?query=...
};

/**
 * Rosca (Sol) endpoints - Pool-based savings system
 */
export const ROSCA_PATHS = {
  // Pools
  POOLS: '/rosca/pools',
  POOL_DETAILS: (poolId: string) => `/rosca/pools/${poolId}`,

  // Enrollments
  ENROLLMENTS: '/rosca/enrollments',
  ENROLLMENT_DETAILS: (enrollmentId: string) => `/rosca/enrollments/${enrollmentId}`,

  // Payments
  PAYMENTS: '/rosca/payments',
  ENROLLMENT_PAYMENTS: (enrollmentId: string) => `/rosca/enrollments/${enrollmentId}/payments`,

  // Friends in pool
  ENROLLMENT_FRIENDS: (enrollmentId: string) => `/rosca/enrollments/${enrollmentId}/friends`,
};

/**
 * Combined export of all API paths
 */
export const API_PATHS = {
  AUTH: AUTH_PATHS,
  USER: USER_PATHS,
  KYC: KYC_PATHS,
  REFERENCE_DATA: REFERENCE_DATA_PATHS,
  ACCOUNT: ACCOUNT_PATHS,
  WALLET: WALLET_PATHS,
  TRANSACTION: TRANSACTION_PATHS,
  INTERACTION: INTERACTION_PATHS,
  MESSAGE: MESSAGE_PATHS,
  BUSINESS: BUSINESS_PATHS,
  NOTIFICATION: NOTIFICATION_PATHS,
  ENTITIES: ENTITY_PATHS,
  SEARCH: SEARCH_PATHS,
  ROSCA: ROSCA_PATHS,
};

export default API_PATHS; 